tests/schema/test_basic.py

```diff
@@ -580,3 +580,43 @@ class FooBar2:
             FooBar(foo=1)
         FooBar(bar=2)
         FooBar(foo=1, bar=2)
+
+
+def test_use_default_factory_on_root():
+    @strawberry.type
+    class Sub:
+        foo: str = strawberry.field(default="bar")
+
+    @strawberry.type
+    class Query:
+        sub: Sub = strawberry.field(default_factory=Sub)
+
+    schema = strawberry.Schema(query=Query)
+
+    result = schema.execute_sync(
+        """
+        query TestQuery{
+            sub {
+                foo
+            }
+        }
+        """
+    )
```

Comment:
isn't this fixed by passing a root value to schema.execute? 😊

Reply:
sounds good. There should be a default root value in execute instead of None

Reply:
I get you, but I think that would make us quite opinionated, and also we can't guess how all types will be instantiated, this would only work if types don't require any argument 😊

Reply:
This task isn't easy as you have to provide a mutation or query instance to execute.
this is a big task